File: java/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int res = 0;
        int n = mat.length;

        for (int i = 0; i < n; i++) {
            res += mat[i][i];
            res += mat[i][n - 1 - i];
        }

        return res - (n % 2 == 1 ? mat[n / 2][n / 2] : 0);
    }
}

File: kotlin/1572-matrix-diagonal-sum.kt
class Solution {
    fun diagonalSum(mat: Array<IntArray>): Int {
        val n = mat.lastIndex

        var sum = 0
        for (i in 0..n) {
            sum += mat[i][i] + mat[n - i][i]
        }

        if (n % 2 == 0)
            sum -= mat[n/2][n/2]

        return sum
    }
}
